pkg/controller/applications/controller.go

```diff
@@ -0,0 +1,241 @@
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) { // nolint:gocyclo
+	cr, ok := mg.(*v1alpha1.Application)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotApplication)
+	}
+
+	createRequest := generateCreateApplicationRequest(&cr.Spec.ForProvider, cr.Name)
+
+	_, err := e.client.Create(ctx, createRequest)
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateFailed)
+	}
+
+	return managed.ExternalCreation{}, errors.Wrap(nil, errKubeUpdateFailed)
+}
+
+func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) { // nolint:gocyclo
+	cr, ok := mg.(*v1alpha1.Application)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotApplication)
+	}
+	name := meta.GetExternalName(cr)
+	updateRequest := generateUpdateRepositoryOptions(&cr.Spec.ForProvider, name)
+	_, err := e.client.Update(ctx, updateRequest)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errUpdateFailed)
+	}
+
+	return managed.ExternalUpdate{}, nil
+}
+
+func (e *external) Delete(ctx context.Context, mg resource.Managed) error {
+	cr, ok := mg.(*v1alpha1.Application)
+	if !ok {
+		return errors.New(errNotApplication)
+	}
+	query := application.ApplicationDeleteRequest{
+		Name: clients.StringToPtr(meta.GetExternalName(cr)),
+	}
+
+	_, err := e.client.Delete(ctx, &query)
+
+	return errors.Wrap(err, errDeleteFailed)
+}
+
+func lateInitialize(applicationParameters *v1alpha1.ApplicationParameters, app *argocdv1alpha1.Application) { // nolint:gocyclo
+	if app == nil {
+		return
+	}
+	if applicationParameters == nil {
+		return
+	}
+	// To be considered in future
+}
+
+func generateApplicationObservation(app *argocdv1alpha1.Application) v1alpha1.ArgoApplicationStatus {
+	if app == nil {
+		return v1alpha1.ArgoApplicationStatus{}
+	}
+
+	converter := v1alpha1.ConverterImpl{}
+	status := converter.FromArgoApplicationStatus(&app.Status)
+	return *status
+}
+
+func generateCreateApplicationRequest(p *v1alpha1.ApplicationParameters, name string) *application.ApplicationCreateRequest { // nolint:gocyclo
+	converter := v1alpha1.ConverterImpl{}
+	spec := converter.ToArgoApplicationSpec(p)
+
+	app := &argocdv1alpha1.Application{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: *spec,
+	}
+
+	repoCreateRequest := &application.ApplicationCreateRequest{
+		Application: app,
+	}
+
+	return repoCreateRequest
+}
+
+func generateUpdateRepositoryOptions(cr *v1alpha1.ApplicationParameters, name string) *application.ApplicationUpdateRequest {
+	converter := v1alpha1.ConverterImpl{}
+
+	spec := converter.ToArgoApplicationSpec(cr)
+
+	app := &argocdv1alpha1.Application{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: *spec,
+	}
+
+	o := &application.ApplicationUpdateRequest{
+		Application: app,
+	}
+	return o
+}
```

Comment:
Make it simpler by passing `cr` and get the name automatically:

```suggestion
func generateUpdateRepositoryOptions(cr *v1alpha1.Application) *application.ApplicationUpdateRequest {
	converter := v1alpha1.ConverterImpl{}

	spec := converter.ToArgoApplicationSpec(&cr.Spec.ForProvider)

	app := &argocdv1alpha1.Application{
		TypeMeta: metav1.TypeMeta{},
		ObjectMeta: metav1.ObjectMeta{
			Name: meta.GetExternalName(cr),
		},
		Spec: *spec,
	}

	o := &application.ApplicationUpdateRequest{
		Application: app,
	}
	return o
}
```